excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/CaseTypeParser.java

```diff
@@ -87,32 +104,38 @@ public CaseTypeParser(ParseContext parseContext,
                 .addSearchAliasFields(searchAliasFieldParser.parseAll(definitionSheets, caseType));
 
             parseContext.registerCaseRoles(new ArrayList<>(caseRoleEntities));
-            caseType.addCaseTypeACLEntities(authorisationCaseTypeParser.parseAll(definitionSheets, caseType));
+            Collection<CaseTypeACLEntity> caseTypeACLEntities = authorisationCaseTypeParser.parseAll(definitionSheets, caseType);
+            caseType.addCaseTypeACLEntities(caseTypeACLEntities);
+            caseTypeACLEntities.stream().filter(entity -> entity.getUserRole() == null).forEach(entity -> missingUserRoles.add(entity.getUserRoleId()));
```

Comment:
as an alternative, can't we have a `missingUserRoles` initialised as empty inside the ParseContext, and all the various authorisation parsers when parsing a row adding to `missingUserRoles` if needed? Then this method would just return `missingUserRoles` as part of the ParseResult.

Reply:
I agree.. I will change this...